packages/main/src/controllers/message/MessageService.ts

```diff
@@ -235,6 +238,37 @@ export class MessageService {
             )
           }
         }
+      } else if (messageType === CredentialRevocationMessage.type) {
+        const msg = JsonTransformer.fromJSON(message, CredentialRevocationMessage)
+
+        const credentials = await agent.credentials.findAllByQuery({ threadId: msg.threadId })
+        if (credentials && credentials.length > 0) {
+          for (const credential of credentials) {
+            const isRevoke = Boolean(
+              credential.getTag('anonCredsRevocationRegistryId') &&
+                credential.getTag('anonCredsCredentialRevocationId'),
+            )
+            isRevoke &&
```

Comment:
I don't like this React-style way of doing conditional statements in Node.

Also you can do the same thing a lot better by simply adding a line that checks for revocation and otherwise throws error. e.g..

```
if (!isRevocable) throw new Error('Credential for threadId ${msg.threadId} is not revocable)

await agent.modules.anoncreds.updateRevocationStatusList(...)
await agent.credentials.sendRevocationNotification(...)
...
```